Update ExportPlaylists.applescript

pk2061 opened the pull request:
Modified the script to make use of relative paths in the m3u-file.

TODO: 
Currently the m3u always contains relative paths. Maybe there is a way to incorporate a simple solution to make this optional.

pk2061 committed:
Update ExportPlaylists.applescript

Modified the script to make use of relative paths in the m3u-file.

pk2061 commented:
Also: Changed to run with the Music-App instead of iTunes under Catalina / BigSur.

pk2061 committed:
Update README.md

dpet23 commented on README.md:
```diff
@@ -38,10 +38,10 @@ Created to easily export playlists and media from iTunes to Android.
 | --- | --- |
 | `make` | Build an app, and save it to this directory |
-| `make deploy` | Build an app, and move it to `~/Library/iTunes/Scripts/` |
+| `make deploy` | Build an app, and move it to `~/Library/Music/Scripts/` |
```
The `Makefile` hasn't been updated, and it will still move the app to `~/Library/iTunes/Scripts/`.

The current `deploy` rule could be split into `deploy-itunes` and `deploy-music`.

pk2061 replied:
I splitted the existing deploy rule into deploy rules for iTunes and Music. 
The clean-deploy rule should now delete the app file from booth /Library/iTunes/Scripts/ and /Library/Music/Scripts/ExportPlaylists.app

dpet23 commented on README.md:
```diff
@@ -38,10 +38,10 @@ Created to easily export playlists and media from iTunes to Android.
 | `make exportplaylists` | Build an app from the `ExportPlaylists` script |
 | `make clean` | Remove all apps from this directory |
-| `make clean-deploy` | Remove the app from `~/Library/iTunes/Scripts/` |
+| `make clean-deploy` | Remove the app from `~/Library/Music/Scripts/` |
```
The `Makefile` will still remove the app from `~/Library/iTunes/Scripts/`.

The `clean-deploy` rule could be updated to delete from both the `iTunes` and `Music` directories, ignoring any errors (eg. if the `iTunes` folder doesn't exist).

pk2061 replied:
I updated the README.md to reference both iTunes and Music accordingly with their methods of integration etc.

dpet23 commented on ExportPlaylists.applescript:
```diff
@@ -500,7 +503,16 @@ with timeout of 60 * 60 * 24 seconds -- (timeout of 24 hours for many huge playl
+								-- CHECK FOR RELATIVE PATH
+								if (true) then -- TODO: ADD option for relative path:
+									
+									set cwd to "./Music/"
```
When exporting multiple playlists to the same folder, the `./Music/` path will be invalid. The path here should be the relative path of `musicPath` compared to `playlistsPath`.

pk2061 replied:
I changed the application back to iTunes to keep backward compatibility.

I added global variables for global musicFolder_DifferentFolder, global musicFolder_SameFolder and global playlistFolder_SameFolder to make adjustments to the default folders which are created during the export process more easier in the future.

I fixed the bug concerning the exporting of multiple playlists with the "Same folder" option by adjusting the relative path written to the playlist file.

dpet23 reviewed (changes requested):
Good job overall, and thank you for contributing!

There's a few changes I'd like to see before merging.

pk2061 committed:
Updated Makefile & README.md
Rewrote ExportPlaylist to fix a bug with multiple playlist export

pk2061 committed:
Makefile:
- Added deployoption for iTunes or Music
 README.md:
 - Added description related to the inegration into Music
  ExportPlaylists.applescript
- Rewrote part of the definition an handling of default folders and fixed a bug relating to the export of multiple playlists with the "Same folder" option.

pk2061 committed:
Fixed messed up README.md
